File: outdoor-saas-be/src/main/java/com/touhuwai/service/FrameService.java
package com.touhuwai.service;

import com.touhuwai.common.PageResult;
import com.touhuwai.entity.Frame;

import java.util.List;

/**
 * 框架媒体服务接口
 */
public interface FrameService {
    
    /**
     * 根据ID查询框架
     * @param id 框架ID
     * @return 框架信息
     */
    Frame getById(Integer id);
    
    /**
     * 根据框架编号查询
     * @param frameNo 框架编号
     * @return 框架信息
     */
    Frame getByFrameNo(String frameNo);
    
    /**
     * 查询所有框架
     * @return 框架列表
     */
    List<Frame> getAll();
    
    /**
     * 分页查询框架列表
     * @param frame 查询条件
     * @param pageNum 页码
     * @param pageSize 每页大小
     * @return 分页结果
     */
    PageResult<Frame> getPage(Frame frame, Integer pageNum, Integer pageSize);
    
    /**
     * 根据社区ID查询框架列表
     * @param communityId 社区ID
     * @return 框架列表
     */
    List<Frame> getByCommunityId(Integer communityId);
    
    /**
     * 新增框架
     * @param frame 框架信息
     * @return 影响的行数
     */
    int add(Frame frame);
    
    /**
     * 批量新增框架
     * @param list 框架列表
     * @return 影响的行数
     */
    int batchAdd(List<Frame> list);
    
    /**
     * 更新框架
     * @param frame 框架信息
     * @return 影响的行数
     */
    int update(Frame frame);
    
    /**
     * 根据ID删除框架
     * @param id 框架ID
     * @return 影响的行数
     */
    int delete(Integer id);
    
    /**
     * 批量删除框架
     * @param ids ID列表
     * @return 影响的行数
     */
    int batchDelete(List<Integer> ids);
}

File: outdoor-saas-fe/src/services/frameService.ts
/**
 * 框架媒体管理服务
 */
import request from './api';
import type { FrameQueryParam, PageResult } from '../types/query';

export interface Frame {
  id: number;
  frameNo: string;
  communityId: number;
  communityName?: string;
  building: string;
  unit: string;
  elevator: string;
  innerPosition?: number;
  releaseDateBegin?: string;
  releaseDateEnd?: string;
  releaseStatus?: number;
  createdAt?: string;
  updatedAt?: string;
}

export const frameService = {
  // 获取所有框架
  getAll: () => request<Frame[]>('/frame/list'),

  // 根据ID获取框架
  getById: (id: number) => request<Frame>(`/frame/${id}`),

  // 根据框架编号获取
  getByNo: (frameNo: string) => request<Frame>(`/frame/no/${frameNo}`),

  // 分页查询框架
  getPage: (page: number = 1, size: number = 10) =>
    request<Frame[]>('/frame/page', {
      method: 'POST',
      body: JSON.stringify({ page, size }),
    }),

  // 根据社区ID获取框架
  getByCommunity: (communityId: number) =>
    request<Frame[]>(`/frame/community/${communityId}`),

  // 创建框架
  create: (data: Partial<Frame>) =>
    request<Frame>('/frame', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  // 更新框架
  update: (data: Partial<Frame>) =>
    request<Frame>('/frame', {
      method: 'PUT',
      body: JSON.stringify(data),
    }),

  // 删除框架
  delete: (id: number) =>
    request<void>(`/frame/${id}`, { method: 'DELETE' }),

  // 批量删除框架
  deleteBatch: (ids: number[]) =>
    request<void>('/frame/batch', {
      method: 'DELETE',
      body: JSON.stringify(ids),
    }),

  // 过滤查询框架列表（不分页）
  filter: (params: FrameQueryParam) =>
    request<PageResult<Frame>>('/frame/filter', {
      method: 'POST',
      body: JSON.stringify(params),
    }),

  // 过滤查询框架列表（分页）
  filterPage: (params: FrameQueryParam) =>
    request<PageResult<Frame>>('/frame/filter/page', {
      method: 'POST',
      body: JSON.stringify(params),
    }),
};

export default frameService;
